openwisp_controller/connection/api/serializer.py

```diff
@@ -10,7 +10,15 @@
 Device = load_model('config', 'Device')
 
 
-class CommandSerializer(serializers.ModelSerializer):
+class DeviceValidationMixin(object):
+    def validate(self, data):
+        data['device'] = Device.objects.get(pk=self.context['device_id'])
+        instance = self.instance or self.Meta.model(**data)
+        instance.full_clean()
+        return data
```

Comment:
I don't understand what this mixin is doing exactly and differently from `ValidatedModelSerializer`?
Also why can't you inherit this mixin from `ValidatedModelSerializer` so to avoid writing both below?

Reply:
It is required for adding `device_id` to the data. The `device_id` is taken from the URL kwarg of the view and passed in the serializer context.

Reply:
thanks for adding the comment